### DISCUSSION TITLE
https://mudmapbuilder.github.io/

### DISCUSSION THREAD

**Discussion by rds1983** [malkamius/CrimsonStainedLands#6] [Show and tell]
Hello,
This page has maps and eqlists of various open source mud codebases.
Recently I've added CSL there too.
https://mudmapbuilder.github.io/

**Comment by malkamius** [comment]
The resource site and github repositories have been added to the readme. Your site has been linked to from the CSL site too.

**Reply by malkamius** [reply]
https://github.com/malkamius/CSLMapEditor

Possible start of a map editor which uses the CSL mud engine to load all the area xml files. Definitely not done, but I haven't spent any time working on it recently. You're supposed to be able to add/edit items, npcs and rooms with it, but it's probably not bug free.

**Reply by rds1983** [reply]
Nice
Btw, I've noticed that Equip and Give reset items have Destination set to zero.
I.e. https://github.com/malkamius/CrimsonStainedLands/blob/65c8e1a10f78d98cc643b944afdbce0e7ad8d88f/master/CrimsonStainedLands/data/areas/Astoria%20(New%20Hometown).xml#L11654

Is it a bug? Shouldn't it be set to the vnum of the mobile?

**Reply by malkamius** [reply]
I believe give and equip target the last reset npc

**Reply by rds1983** [reply]
oh, I see

**Reply by malkamius** [reply]
Yes, this allows for resetting the same NPC but having different equipment on different "instances"

Also, the reset system has a EquipRandom mode where it can choose from a list of items to give the NPC. It is only used on my private server right now though.

**Comment by malkamius** [comment]
![image](https://github.com/malkamius/CrimsonStainedLands/assets/51098637/ff673ab1-26e1-4276-ab0f-ae0ae6cb862f)
My in game map command puts points of interest(shops, guild masters, trainers) on the map too, could something like that be possible?

**Reply by rds1983** [reply]
Well, any text can be put inside the boxes representing rooms.
The lib that imports CSL's xml files is this one: https://github.com/rds1983/DikuLoad/tree/master/DikuLoad.Import.CSL
Feel free to make PR that generated more sophisticated room names.
